src/autoType.js

```diff
@@ -1,14 +1,20 @@
 export default function autoType(object) {
   for (var key in object) {
-    var value = object[key].trim(), number;
+    var value = object[key].trim(), number, m;
     if (!value) value = null;
     else if (value === "true") value = true;
     else if (value === "false") value = false;
     else if (value === "NaN") value = NaN;
     else if (!isNaN(number = +value)) value = number;
-    else if (/^([-+]\d{2})?\d{4}(-\d{2}(-\d{2})?)?(T\d{2}:\d{2}(:\d{2}(\.\d{3})?)?(Z|[-+]\d{2}:\d{2})?)?$/.test(value)) value = new Date(value);
+    else if (m = value.match(/^([-+]\d{2})?\d{4}(-\d{2}(-\d{2})?)?(T\d{2}:\d{2}(:\d{2}(\.\d{3})?)?(Z|[-+]\d{2}:\d{2})?)?$/)) {
+      if (fixtz && !!m[4] && !m[7]) value = value.replace(/-/g, "/").replace(/T/, " ");
+      value = new Date(value);
+    }
```

Comment:
How about instead of match and then pulling out the groups, just have a separate regex specific to the workaround?

```js
else if (fixtz && /^([-+]\d{2})?\d{4}(-\d{2}(-\d{2})?)?T\d{2}:\d{2}(:\d{2}(\.\d{3})?)?$/.test(value)) value = new Date(value.replace(/-/g, "/").replace(/T/, " "))
else if (/^([-+]\d{2})?\d{4}(-\d{2}(-\d{2})?)?(T\d{2}:\d{2}(:\d{2}(\.\d{3})?)?(Z|[-+]\d{2}:\d{2})?)?$/.test(value)) value = new Date(value);
```

Reply:
Is it faster? Surprisingly it adds 74 bytes to the gz file size.

Reply:
Either way.

Reply:
For the record, test does appear slightly faster for if the workaround is not needed, but slower if it is (since two tests are needed). But probably in neither case is the difference significant enough to matter. https://observablehq.com/d/e61a03e5eae27be5